modules/gapi/src/backends/streaming/gstreamingbackend.cpp

```diff
@@ -0,0 +1,217 @@
+    >;
+
+using ConstStreamingGraph = ade::ConstTypedGraph
+    < cv::gimpl::Op
+    , StreamingCreateFunction
+    >;
+
+
+class GStreamingIntrinExecutable final: public cv::gimpl::GIslandExecutable
+{
+    virtual void run(std::vector<InObj>  &&,
+                     std::vector<OutObj> &&) override {
+        GAPI_Assert(false && "Not implemented");
+    }
+
+    virtual void run(GIslandExecutable::IInput &in,
+                     GIslandExecutable::IOutput &out) override;
+
+    virtual bool allocatesOutputs() const override { return true; }
+    virtual cv::RMat allocate(const cv::GMatDesc&) const override { return {}; }
+
+    virtual bool canReshape() const override { return true; }
+    virtual void reshape(ade::Graph& graph, const cv::GCompileArgs&) override {
+        // The whole graph here consists only of 1 operation with 1 input node and 1 output one
+        cv::gimpl::GModel::ConstGraph new_gm(graph);
+
+        for (const auto& node_new : graph.nodes())
+        {
+            if (new_gm.metadata(node_new).get<cv::gimpl::NodeType>().t == cv::gimpl::NodeType::DATA) {
+                auto data_new = new_gm.metadata(node_new).get<cv::gimpl::Data>();
+                for (auto&& node : m_g.nodes()) {
+                    if (m_gm.metadata(node).get<cv::gimpl::NodeType>().t == cv::gimpl::NodeType::DATA) {
+                        auto data_old = m_gm.metadata(node).get<cv::gimpl::Data>();
+                        if (data_new.rc == data_old.rc && data_new.shape == data_old.shape) {
+                            data_old.meta = data_new.meta;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+public:
+    GStreamingIntrinExecutable(const ade::Graph                   &,
+                               const std::vector<ade::NodeHandle> &);
+
+    const ade::Graph& m_g;
+    cv::gimpl::GModel::ConstGraph m_gm;
+    cv::gapi::streaming::IActor::Ptr m_actor;
+};
+
+void GStreamingIntrinExecutable::run(GIslandExecutable::IInput  &in,
+                                     GIslandExecutable::IOutput &out)
+{
+    m_actor->run(in, out);
+}
+
+class GStreamingBackendImpl final: public cv::gapi::GBackend::Priv
+{
+    virtual void unpackKernel(ade::Graph            &graph,
+                              const ade::NodeHandle &op_node,
+                              const cv::GKernelImpl &impl) override
+    {
+        StreamingGraph gm(graph);
+        const auto &kimpl  = cv::util::any_cast<cv::gapi::streaming::GStreamingKernel>(impl.opaque);
+        gm.metadata(op_node).set(StreamingCreateFunction{kimpl.createActorFunction});
+    }
+
+    virtual EPtr compile(const ade::Graph &graph,
+                         const cv::GCompileArgs &,
+                         const std::vector<ade::NodeHandle> &nodes) const override
+    {
+        return EPtr{new GStreamingIntrinExecutable(graph, nodes)};
+    }
+
+    virtual bool controlsMerge() const override
+    {
+        return false;
```

Comment:
actually, `true`. Otherwise, `copy(copy)` will result into a two-node island which is not supported by this backend.

Reply:
Fixed